src/liballoc/boxed.rs

```diff
@@ -95,7 +96,7 @@ use crate::str::from_boxed_utf8_unchecked;
 #[lang = "owned_box"]
 #[fundamental]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct Box<T: ?Sized>(Unique<T>);
+pub struct Box<T: ?Sized, A: Alloc + Default = Global>(Unique<T>, PhantomData<A>);
```

Comment:
@scottjmaddox It's fine if we make the `A` parameter unstable:

```suggestion
pub struct Box<
    T: ?Sized,
    #[unstable(feature = "allocator_api", issue = "32838")] A: Alloc = Global,
>(Unique<T>, A);
```

Reply:
Wait, unstable generics arguments are a thing?

Reply:
That would solve a lot of problems!

Reply:
Even without the unstable parameter it wouldn't be possible to write `Box<T, Global>` on stable as `Global` is also unstable. However one could write `Box<T, System>` without creating one.

Reply:
As per https://github.com/rust-lang/wg-allocators/issues/2#issuecomment-488992802 this actually doesn't work :(